packages/terra-date-picker/src/react-datepicker/day.jsx

```diff
@@ -219,7 +219,7 @@ class Day extends React.Component {
     return ['react-datepicker-day', dayClassName, 'react-datepicker-day--' + getDayOfWeekCode(this.props.day), {
       'react-datepicker-day--disabled': this.isDisabled(),
       'react-datepicker-day--selected': this.isSameDay(this.props.selected) && !isDayDisabled(this.props.selected, this.props),
-      'react-datepicker-day--selected-border': this.isSameDay(this.props.preSelection) && !isDayDisabled(this.props.selected, this.props) && (document.activeElement.tagName === 'DIV' || document.activeElement === document.querySelector('[class*="previous"]')) && this.props.isCalendarKeyboardFocused,
+      'react-datepicker-day--selected-border': this.isSameDay(this.props.preSelection) && !isDayDisabled(this.props.selected, this.props) && (document.activeElement.hasAttribute('data-terra-open-calendar-button') || document.activeElement.tagName === 'DIV' || document.activeElement === document.querySelector('[class*="previous"]')) && this.props.isCalendarKeyboardFocused,
```

Comment:
In small breakpoints when you open the picker in portal mode with keyboard, the day is correctly styled with the dotted focus style. If you dismiss the portal picker and click on the calendar button to open it again, it continues to have the focus style applied to the day. If you then resize to a larger breakpoint and open the popup picker by clicking the calendar button, again the day continues to have the focus style when it shouldn't.

Reply:
Fixed 111fd4c

Reply:
The logic on line 222 is getting a bit long and hard to understand. Could we possible wrap some of that in a function to make it easier to understand the logic?

Reply:
Done e892b1d